src/acl/mock.internal.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { IriString, SolidDataset, WithResourceInfo } from "../interfaces";
+import { DataFactory } from "../rdfjs";
+import { Access, AclDataset } from "./acl";
+
+export function mock_addAclRuleQuads(
+  aclDataset: SolidDataset & WithResourceInfo,
+  agent: IriString,
+  resource: IriString,
+  access: Access,
+  type: "resource" | "default" | "legacyDefault",
+  ruleIri?: IriString,
+  targetType:
+    | "http://www.w3.org/ns/auth/acl#agent"
+    | "http://www.w3.org/ns/auth/acl#agentGroup"
+    | "http://www.w3.org/ns/auth/acl#agentClass"
+    | "http://www.w3.org/ns/auth/acl#origin" = "http://www.w3.org/ns/auth/acl#agent"
```

Comment:
Can you not move all these strings into the `Constants.ts` file, and then also make them IRI's (so you don't have to wrap in `DataFactory.namedNode()` calls throughout this file...? In other words, it seems we have some terms in the Constant file, but not all...?

Reply:
I can, but this goes back to the whole DRY/DAMP question: should be the constants used in code and the constants used in the tests reference the same object ? Because what we eventually might want to do is to no longer use the `constants.ts` file, and have all the terms defined in a library we depend on. At this point, isn't it better to have the constants local to the test files for a sanity check, in case the library ever introduces a breaking change that we might want to handle in `solid-client` rather than directly cascading it, possibly missing the effects on our own dependants ?

Reply:
No, I've never in my life seen where it would ever make sense to have test-constants and code-constants separated as a sanity check. The idea is that these values are defined by vocabulary owners outside our control - so yeah, we need a discussion about versioning vocabs, but that's a completely different topic. So yeah, I think having both our code and tests depending on external libraries is clearly where we want to go here (and not 'eventually' either, by 'soon' :) !).